src/bridging/providers/across/AcrossBridgeProvider.ts

```diff
@@ -0,0 +1,271 @@
+    const intermediateToken = getTokenAddress(targetTokenSymbol, sourceChainConfig)
+    return intermediateToken ? [intermediateToken] : []
+  }
+
+  async getQuote(request: QuoteBridgeRequest): Promise<AcrossQuoteResult> {
+    const { sellTokenAddress, sellTokenChainId, buyTokenChainId, amount, recipient } = request
+
+    const suggestedFees = await this.api.getSuggestedFees({
+      token: sellTokenAddress,
+      // inputToken: sellTokenAddress,
+      // outputToken: buyTokenAddress,
+      originChainId: sellTokenChainId,
+      destinationChainId: buyTokenChainId,
+      amount,
+      recipient,
+    })
+
+    console.log('suggestedFees', suggestedFees)
+
+    // TODO: The suggested fees contain way more information. As we review more bridge providers we should revisit the
+    // facade of the quote result.
+    //
+    // For example, this contains also information on the limits, so you don't need to quote again for the same pair.
+    // potentially, this could be cached for a short period of time in the SDK so we can resolve quotes with less
+    // requests.
+
+    return toBridgeQuoteResult(amount, suggestedFees)
+  }
+
+  async getUnsignedBridgeTx(request: QuoteBridgeRequest, quote: AcrossQuoteResult): Promise<EvmCall> {
+    const { sellTokenChainId, sellTokenAddress, buyTokenChainId, buyTokenAddress, owner, recipient } = request
+
+    // Create SpokePool contract
+    const spokePoolAddress = ACROSS_SPOOK_CONTRACT_ADDRESSES[sellTokenChainId]
+    if (!spokePoolAddress) {
+      throw new Error('Spoke pool address not found for chain: ' + sellTokenChainId)
+    }
+    const spokePoolContract = WeirollContract.createContract(
+      new EthersContract(spokePoolAddress, ACROSS_SPOKE_POOL_ABI),
+      CommandFlags.CALL
+    )
+
+    // Create Math contract
+    const mathContractAddress = ACROSS_MATH_CONTRACT_ADDRESSES[sellTokenChainId]
+    if (!mathContractAddress) {
+      throw new Error('Math contract address not found for chain: ' + sellTokenChainId)
+    }
+    const mathContract = WeirollContract.createContract(
+      new EthersContract(mathContractAddress, ACROSS_MATH_ABI),
+      CommandFlags.CALL
+    )
+
+    // Create 'balanceOf' static call contract for the sell token
+    const balanceOfSellTokenContract = WeirollContract.createContract(
+      new EthersContract(sellTokenAddress, ERC20_BALANCE_OF_ABI),
+      CommandFlags.STATICCALL
+    )
+
+    // Create 'approve' call contract for the sell token
+    const approveSellTokenContract = WeirollContract.createContract(
+      new EthersContract(sellTokenAddress, ERC20_APPROVE_OF_ABI),
+      CommandFlags.CALL
+    )
+
+    // Get the cow shed account
+    const cowShedAccount = this.cowShedSdk.getCowShedAccount(sellTokenChainId, owner)
+
+    const { suggestedFees } = quote
+
+    // Get the weiroll call of the deposit into spoke pool
+    const depositCall = createWeirollDelegateCall((planner) => {
```

Comment:
Nitpick.
The `createWeirollDelegateCall` callback might be separated to a function